schema/me/conversation/send_message_mutation.js

```diff
@@ -0,0 +1,82 @@
+import impulse from "lib/loaders/impulse"
+import gravity from "lib/loaders/gravity"
+import { GraphQLString, GraphQLNonNull } from "graphql"
+import { mutationWithClientMutationId, cursorForObjectInConnection } from "graphql-relay"
+import { ConversationType, MessageEdge } from "./index"
+
+const { IMPULSE_APPLICATION_ID } = process.env
+
+const getImpulseToken = gravityToken => {
+  return gravity.with(gravityToken, { method: "POST" })("me/token", {
+    client_application_id: IMPULSE_APPLICATION_ID,
+  })
+}
+
+export default mutationWithClientMutationId({
+  name: "SendConversationMessageMutation",
+  description: "Appending a message to a conversation thread",
+  inputFields: {
+    id: {
+      type: new GraphQLNonNull(GraphQLString),
+      description: "The id of the conversation to be updated",
+    },
+    from: {
+      type: new GraphQLNonNull(GraphQLString),
+      description: "The email address of the message sender",
+    },
+    body_text: {
+      type: new GraphQLNonNull(GraphQLString),
+    },
+    reply_to_message_id: {
+      type: new GraphQLNonNull(GraphQLString),
+      description: "The message being replied to",
+    },
+  },
+  outputFields: {
+    conversation: {
+      type: ConversationType,
+      resolve: ({ conversation }) => conversation,
+    },
+    messageEdge: {
+      type: MessageEdge,
+      resolve: ({ newMessagePayload }) => {
+        return {
+          cursor: cursorForObjectInConnection([newMessagePayload], newMessagePayload),
+          node: newMessagePayload,
+        }
+      },
+    },
+  },
+  mutateAndGetPayload: ({ id, from, to, body_text, reply_to_message_id }, request, { rootValue: { accessToken } }) => {
+    if (!accessToken) return null
+    let impulseToken
+    return getImpulseToken(accessToken)
+      .then(data => {
+        impulseToken = data.token
+        return impulse.with(impulseToken, { method: "POST" })(`conversations/${id}/messages`, {
+          reply_all: true,
+          reply_to_message_id,
+          from,
+          body_text,
+        })
+      })
+      .then(({ id: newMessageID }) => {
+        return impulse.with(impulseToken, { method: "GET" })(`conversations/${id}`).then(updatedConversation => {
```

Comment:
This is from above- why/do we need to refetch the conversation? I'm not sure anything about the conversation will have been updated even after this POST (since it must wait for Radiation callbacks).

If we need to return a new message payload with some ref to the parent conversation, could that just be passed in as an input var and then transformed below (where we transform `newMessagePayload` already).